### PR TITLE
Fabrykant lab03

### PR DESCRIPTION
**Pull request by fabricant**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#include <QDebug>
+
+QString findLongestName(QLinkedList<QString> names);
+QString findShortestName(QLinkedList<QString> names);
+QStringList printLynn(QLinkedList<QString> names);
+void printInBackOrder(QLinkedList<QString> names);
+int main(int argc, char *argv[])
+{
+    QCoreApplication a(argc, argv);
+    QLinkedList<QString> names;
```
Comment (smay1613) [inline main.cpp]: Wrong list type

File: main.cpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    printInBackOrder(names);
+    return a.exec();
+}
+
+
+QString findLongestName(QLinkedList<QString> names) {
+    QString returnString;
+    QLinkedListIterator<QString> iterator(names);
+    while (iterator.hasNext()) {
+        QString next = iterator.next();
```
Comment (smay1613) [inline main.cpp]: "current" is more suitable name

File: main.cpp [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        if (next.length() > returnString.length()) {
+            returnString = next;
+        }
+    }
+    return returnString;
+}
+
+
+QString findShortestName(QLinkedList<QString> names) {
+    auto iterator = names.begin();
```
Comment (smay1613) [inline main.cpp]: What if empty?

File: main.cpp [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    QString returnString = *iterator;
+    ++iterator;
+
+    while (iterator != names.end()) {
```
Comment (smay1613) [inline main.cpp]: Prefer for loop

File: main.cpp [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        }
+        ++iterator;
+    }
+    return returnString;
+}
+
+QStringList printLynn(QLinkedList<QString> names) {
+    QStringList list;
+    foreach(QString str,  names) {
+        if (str.right(4) == "lynn") {
```
Comment (smay1613) [inline main.cpp]: **wrong function used, please choose another way**

File: main.cpp [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+
+void printInBackOrder(QLinkedList<QString> names) {
+    auto iterator = names.rbegin();
+    while (iterator !=  names.rend()) {
+        qDebug() << *iterator;
+        ++iterator;
+
+    }
+}
```
Comment (smay1613) [inline main.cpp:80]: where is pring using qstack/qlist?

**Comment by fabricant** [comment]
fixed

File: main.cpp [hunk 7]
```diff
@@ -38,36 +39,42 @@ QString findLongestName(QLinkedList<QString> names) {
...
-    auto iterator = names.begin();
-    QString returnString = *iterator;
-    ++iterator;
+QString findShortestName(QList<QString> names) {
+    QString returnString = "";
 
-    while (iterator != names.end()) {
-        if (iterator->length() < returnString.length()) {
-            returnString = *iterator;
+    for (auto current = names.begin(); current < names.end(); ++current) {
```
Comment (smay1613) [inline main.cpp:45]: _take a note that C++11 range for is better_

File: main.cpp [hunk 8]
```diff
@@ -38,36 +39,42 @@ QString findLongestName(QLinkedList<QString> names) {
...
+        qDebug() << *current;
+        ++current;
+
+    }
+
+    qDebug() << "QStack:";
+    QStack<QString> namesStack;
+    namesStack += QStack<QString>::fromList(names);
+    while (namesStack.size()) {
+        qDebug () << namesStack.pop();
```
Comment (smay1613) [inline main.cpp:78]: it is not necessary to pop elements. you can iterate over QStack

**Pull request closed without merging**